test.py

```diff
@@ -114,25 +119,55 @@ def test_encoding(self):
             # Also, none of these should parse when interpreted as UTF-8
             self.assertRaises(Exception, str.decode, req.content, 'utf-8')
 
-
-class ContextManagerTest(unittest.TestCase):
-    '''Testing out using the context manager version'''
-
-    def test_context_manager(self):
-        '''Testing out using the context manager version'''
-        url = 'http://localhost:8080/basic/basic.asis'
-        # Shouldn't be able to fetch
-        self.assertRaises(requests.exceptions.ConnectionError,
-            requests.get, url)
-        with asis.Server('test', port=8080, server='gevent'):
-            # Make sure we can get requests now...
-            req = requests.get(url)
+    def test_charset_no_length(self):
+        '''When a charset is used, but a content-length is absent, none is provided.'''
+        # Just exercise this branch
```

Comment:
Turns out that `bottle` _really_ doesn't want you to return no `Content-Length`.